src/pudl/transform/params/ferc1.py

```diff
@@ -3396,6 +3396,77 @@
             "on": "ferc_account_label",
         },
     },
+    "retained_earnings_ferc1": {
+        "rename_columns_ferc1": {
+            "duration_xbrl": {
+                "columns": {
+                    "retained_earnings_primary_contra_account_affected": "retained_earnings_contra_primary_account_affected"
+                }
+                | {
+                    col: f"{col}_contra_primary_account_affected"
+                    for col in [
+                        "adjustments_to_retained_earnings_debit",
+                        "balance_transferred_from_income",
+                        "appropriations_of_retained_earnings",
+                        "adjustments_to_retained_earnings_credit",
+                        "dividends_declared_common_stock",
+                        "dividends_declared_preferred_stock",
+                        "changes_unappropriated_undistributed_subsidiary_earnings_credits",
+                    ]
+                }
+            },
+            "xbrl": {
+                "columns": {
+                    "entity_id": "utility_id_ferc1_xbrl",
+                    "report_year": "report_year",
+                    "contra_primary_account_affected": "amount",
+                }
+            },
+            "dbf": {
+                "columns": {
+                    "respondent_id": "utility_id_ferc1_dbf",
+                    "report_year": "report_year",
+                    "spplmnt_num": "spplmnt_num",
+                    "row_number": "row_number",
+                    "row_seq": "row_seq",
+                    "row_prvlg": "row_prvlg",
+                    "report_prd": "report_prd",
+                    "xbrl_factoid": "earnings_type",
+                    "prev_balance": "starting_balance",
+                }
+            },
+        },
+        "wide_to_tidy": {
+            "xbrl": {
+                "idx_cols": [
+                    "entity_id",
+                    "report_year",
+                ],
+                "value_types": [
+                    "starting_balance",
+                    "ending_balance",
+                    "contra_primary_account_affected",
+                ],
+                "expected_drop_cols": 2,
+                "stacked_column_name": "earnings_type",
+            },
+        },
+        "align_row_numbers_dbf": {"dbf_table_names": ["f1_retained_erng"]},
+        "select_dbf_rows_from_category": {
+            "column_name": "earnings_type",
+            "select_based_on_xbrl_category": True,
+            "expected_categories_to_drop": 6,
+        },
+        "unstack_balances_to_report_year_instant_xbrl": {
+            "unstack_balances_to_report_year": True
+        },
+        "merge_xbrl_metadata": {
+            "rename_columns": {"xbrl_factoid": "earnings_type"},
+            "on": "earnings_type",
+        },
+        "strip_non_numeric_values": {"amount": {"strip_non_numeric_values": True}},
+    },
+    "retained_earnings_appropriations_ferc1": {},
```

Comment:
It's fine for now if you're going on to the appropriations table immediately, but beyond just creating `TABLE_NAME_MAP` probably these speculative future references to tables that aren't done yet are better to avoid -- in the resource definitions, class / TableId definitions, params here. They could easily become cruft.

Reply:
yeah fair i'll rip em out